simtools/simtel/simtel_config_writer.py

```diff
@@ -71,35 +71,38 @@ def write_telescope_config_file(self, config_file_path, parameters, config_param
             )
             file.write("#endif\n\n")
 
-            self._add_simtel_metadata(parameters, "telescope")
             if config_parameters is not None:
                 parameters.update(config_parameters)
 
-            for par, value in parameters.items():
-                _simtel_name = names.get_simtel_name_from_parameter_name(
-                    par, search_telescope_parameters=True, search_site_parameters=False
-                )
+            for _simtel_name, value in parameters.items():
                 if _simtel_name is not None:
                     value = "none" if value is None else value  # simtel requires 'none'
                     if isinstance(value, bool):
                         value = 1 if value else 0
                     elif isinstance(value, (list, np.ndarray)):
                         value = gen.convert_list_to_string(value)
                     file.write(f"{_simtel_name} = {value}\n")
+            _config_meta = self._add_simtel_metadata("telescope")
+            for _simtel_name, value in _config_meta.items():
+                file.write(f"{_simtel_name} = {value}\n")
 
-    def _add_simtel_metadata(self, parameters, config_type):
+    def _add_simtel_metadata(self, config_type):
```

Comment:
Maybe this should be renamed to `get_simtel_metadata`. Similar to `get_simtel_parameters()` in the following.

Reply:
Good point, changed this to _get_simtel_metadata().